translate/src/hooks/useWindowWidth.ts

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useState } from 'react';
+
+const MAX_WIDTH = {
+  narrow: 600,
+  medium: 800,
+};
+
+/**
+ * Return true if the screen is narrower than the given width.
+ *
+ * Useful in Responsive Web Design.
+ */
+export function useWindowWidth(width: 'narrow' | 'medium'): boolean {
```

Comment:
This hook name is misleading, because `narrow` and `medium` sound like they would be exclusive with each other, but with this implementation `narrow` is a subset of `medium`. That is not how it usually works.

In my earlier proposal, I had this returning effectively an enum of `narrow`/`medium`/`wide`, and for the "narrower than" type of comparison to be done in the calling code.